### DISCUSSION TITLE
Sending HTML output as body in MS Outlook with base64 image corrupts the image.

### DISCUSSION THREAD

**Discussion by steven-peeters1** [eclipse-birt/birt#1100] [General]
Birt 4.4.2
Rendering a file to HTML with an image so it can be used as an e-mailbody in MS Outlook.
The image is initially embedded in the report design, and is converted to a base64 string using an ImageHandler (Birt standalone runtime)
code used: https://forums.opentext.com/forums/developer/discussion/61437/base64-imagehandler

At first glance, the resulting HTML file looks fine when opening in a browser and the images are showing.
Note that the "img" tag is output as one long string (1 line when opened in a source editor)

However if I try to send the file as a body in MS Outlook the images are corrupted.
It seems like, for Outlook, there is a maximum number of characters that can be in one line.
For (larger) images, an exlamation-mark is added at the end of the encoded string, corrupting the image.

I have tried to encode the base64-image as a chuncked string (76 character blocks).
However when I do that, the resulting birt-HTML file wil still show the Base64-string in one line.
It will just add "(ampersand)#13;(ampersand)#10;" after every chunk.

Is there any way around this?
I'd like the resulting HTML-render to actually have a line-break in the source-code, not the HTML-escape variant.
As a last resort I could just put line-breaks in the source file in Birt RE after the HTML-render is done.

**Comment by steven-peeters1** [comment]
RFC 2822 states that "Each line of characters MUST be no more than 998 characters, and SHOULD be no more than 78 characters, excluding the CRLF." If it exceeds this limit, an exclamation mark will be added in the text.
Still not sure if this is done by our sendmail UNIX program that sends the mail, or by outlook that receives it.

Fixed this by applying soms text-functions to the resulting html output-file
- Issue with html-rendered output due to Base64.encodeBase64Chunked used for base64 images -> replace crlf html-entities  in the html-output by actual crlf (\r\n)
- In birt html-output you could, in theory have one line that exceeds the 998 limit -> Use WordUtils.wrap from the Apache commons library to wrap these long lines. This method will prevent any tags to be cut off in the middle of the tag.

**Comment by hvbtup** [comment]
Thx for sharing your experience.
I once tried to send email messages containing HTML with images (not generated with BIRT), and found that it is something which should generally be avoided, because you never know if and how the image is rendered by different email clients.
